http.go

```diff
@@ -18,15 +20,22 @@ func NewHTTPClient(ctx context.Context) *http.Client {
 	return defaultClient
 }
 
-func HTTPRequestExecutor(clientFactory transport.HTTPClientFactory) transport.HTTPRequestExecutor {
+func HTTPRequestExecutor(clientFactory transport.HTTPClientFactory, cfg *config.Backend) transport.HTTPRequestExecutor {
 	if !IsBackendEnabled() {
 		return transport.DefaultHTTPRequestExecutor(clientFactory)
 	}
 	return func(ctx context.Context, req *http.Request) (*http.Response, error) {
-		client := clientFactory(ctx)
-		if _, ok := client.Transport.(*ochttp.Transport); !ok {
-			client.Transport = &ochttp.Transport{Base: client.Transport}
+		// tags for stats
+		tags := []tagGenerator{
+			func(r *http.Request) tag.Mutator { return tag.Upsert(ochttp.KeyClientHost, req.Host) },
+			func(r *http.Request) tag.Mutator { return tag.Upsert(ochttp.Host, req.Host) },
+			func(r *http.Request) tag.Mutator { return tag.Upsert(ochttp.KeyClientPath, GetAggregatedPathForBackendMetrics(cfg, req)) },
+			func(r *http.Request) tag.Mutator { return tag.Upsert(ochttp.Path, GetAggregatedPathForBackendMetrics(cfg, req)) },
+			func(r *http.Request) tag.Mutator { return tag.Upsert(ochttp.KeyClientMethod, req.Method) },
+			func(r *http.Request) tag.Mutator { return tag.Upsert(ochttp.Method, req.Method) },
 		}
+		client := clientFactory(ctx)
+		client.Transport = &Transport{Base: client.Transport, tags: tags}
```

Comment:
You create on every request create a new wrapper for Transport.

```suggestion
		if _, ok := client.Transport.(*Transport); !ok {
 			client.Transport = &Transport{Base: client.Transport, tags: tags}
 		}
```